cli/pkg/validate/entity/property.go

```diff
@@ -0,0 +1,144 @@
+package entity
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rudderlabs/rudder-iac/cli/pkg/localcatalog"
+	"github.com/samber/lo"
+)
+
+var (
+	_ TypedCatalogEntityValidator[*localcatalog.Property] = &PropertyEntityValidator{}
+	_ ValidationRule[*localcatalog.Property]              = &PropertyRequiredKeysRule{}
+	_ ValidationRule[*localcatalog.Property]              = &PropertyDuplicateKeysRule{}
+)
+
+var validPropertyTypes = []string{"string", "number", "boolean", "object", "array"}
+
+type PropertyEntityValidator struct {
+	rules []ValidationRule[*localcatalog.Property]
+}
+
+func (v *PropertyEntityValidator) RegisterRule(rule ValidationRule[*localcatalog.Property]) {
+	v.rules = append(v.rules, rule)
+}
+
+func (pv *PropertyEntityValidator) Validate(dc *localcatalog.DataCatalog) []ValidationError {
+
+	var errors []ValidationError
+
+	for group, properties := range dc.Properties {
+		for _, property := range properties {
+
+			reference := fmt.Sprintf(
+				"#/properties/%s/%s",
+				group,
+				property.LocalID,
+			)
```

Comment:
Something to consider about references, probably in the context of the loader: How can we make references a first class citizen instead of relying different parts of the code having to create them (and repeat the way they do?)